aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -818,6 +843,156 @@ Tensor& linalg_cholesky_out(Tensor &result, const Tensor &self) {
   return result;
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ cholesky_inverse ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/*
+Copies the lower (or upper) triangle of the square matrix to the other half and conjugates it.
+This operation is performed in-place.
+*/
+template <typename scalar_t>
+static inline void apply_reflect_conj_tri_single(scalar_t* self, int64_t n, int64_t stride, bool upper) {
+  std::function<void(int64_t, int64_t)> loop = [](int64_t, int64_t){};
+  if (upper) {
+    loop = [&](int64_t start, int64_t end) {
+      for (int64_t i = start; i < end; i++) {
+        for (int64_t j = i + 1; j < n; j++) {
+          self[i * stride + j] = conj_impl(self[j * stride + i]);
```

Comment:
Is this implementation of conj_impl distinct from the CUDA one?

Reply:
Yes, implementation is a bit different, but they do the same thing. This `conj_impl` is defined here https://github.com/pytorch/pytorch/blob/461aafe389b7b5b415c3b4878abaa0ba1769f132/aten/src/ATen/native/cpu/zmath.h#L120-L133, it can be used only in CPU code.
While the CUDA one `conj_wrapper` defined here https://github.com/pytorch/pytorch/blob/461aafe389b7b5b415c3b4878abaa0ba1769f132/aten/src/ATen/native/cuda/UnaryComplexKernels.cu#L72-L89 is the same as used for `.conj()` on CUDA tensors. Can be used inside device functions because it is decorated with `__device__`.

Reply:
Would the C10_HOST_DEVICE macro help us unify them?

https://github.com/pytorch/pytorch/blob/8aad66a7bd35782a0e662d133276830204499656/c10/macros/Macros.h#L202

cc @ngimel

Reply:
I agree with @mruberry, it seems like we can get rid of `conj_impl`, `angl_impl` etc. if we use `C10_HOST_DEVICE` and define them in a way similar to other functions in `complex_math.h`. @zasdfgbnm was there a reason why we didn't unify `zmath.h` and `complex_math.h`?

Reply:
I think `zmath.h` was a file to put utilities used in implementing CPU kernels and this file is still there only because of historical reasons. I agree that we can get rid of these `conj_impl` and `std::conj` should be working, `angl_impl` as well.

Reply:
`std::conj` doesn't work here directly because it casts `double`/`float` to complex and always returns complex number, while `conj_impl` does nothing for `double`/`float`.

Reply:
I think you're right, @IvanYashchuk, that we need conj_impl != std::conj. We could also file an issue to follow-up on the question of whether these implementations could be merged. We don't need to hold this PR up for that discussion.

Reply:
The ask here is not to have a single `std::conj` call for everything, but to have a single wrapper for cpu and cuda code instead of conj_wrapper and conj_impl (e.g. on cuda there's conj_wrapper specialized for complex and for everything else).

Reply:
Brief summary of offline conversation:

We would like to look at resolving whether conj_impl and conj_wrapper can be merged in this PR.

Reply:
`conj_wrapper` is not used anymore in this PR. I removed the cuda kernel that used it because cuda implementation is based on another function now.